File: apps/ios/M0ProtocolMockClient/Sources/M0ProtocolMockClient/Models/M0ResumeRequest.swift
import Foundation

/// Request metadata used to reconnect and replay from a known stream cursor.
public struct M0ResumeRequest: Codable, Equatable, Sendable {
    /// Identifier of the stream that should be resumed.
    public let streamID: String

    /// Sender sequence value for the reconnect request itself.
    public let streamSeq: UInt64

    /// Last sequence already processed by the reconnecting client.
    public let lastSeq: UInt64

    /// Creates a resume request.
    /// - Parameters:
    ///   - streamID: Stream identifier to resume.
    ///   - streamSeq: Sequence assigned to the reconnect request.
    ///   - lastSeq: Last processed event sequence.
    public init(streamID: String, streamSeq: UInt64, lastSeq: UInt64) {
        self.streamID = streamID
        self.streamSeq = streamSeq
        self.lastSeq = lastSeq
    }

    private enum CodingKeys: String, CodingKey {
        case streamID = "stream_id"
        case streamSeq = "stream_seq"
        case lastSeq = "last_seq"
    }
}

File: apps/android/M0ProtocolMockClient/src/main/kotlin/com/commandrelay/m0protocol/models/M0ResumeRequest.kt
package com.commandrelay.m0protocol.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Metadata used to reconnect and replay from a known stream cursor.
 *
 * @property streamId Identifier of the stream that should be resumed.
 * @property streamSeq Sender sequence value for the reconnect request itself.
 * @property lastSeq Last sequence already processed by the reconnecting client.
 */
@Serializable
public data class M0ResumeRequest(
    @SerialName("stream_id")
    public val streamId: String,
    @SerialName("stream_seq")
    public val streamSeq: Long,
    @SerialName("last_seq")
    public val lastSeq: Long,
)
